src/StrictPhp/TypeChecker/TypeChecker/TypedTraversableChecker.php

```diff
@@ -79,7 +80,14 @@ public function simulateFailure($value, Type $type)
             ));
         }
 
+        $subType = $type->getValueType();
+
         if (! $value instanceof \Traversable) {
+
+            if (!is_array($value) && !$subType instanceof Mixed) {
```

Comment:
This should actually work, as we have a `MixedTypeChecker` that accepts this sort of type now